src/frontend/src/optimizer/plan_node/logical_scan.rs

```diff
@@ -692,9 +727,12 @@ impl ToStream for LogicalScan {
         if columns.is_empty() {
             return None;
         }
+
         if self.table_indexes().is_empty() {
+            self.notice_recommended_index(columns);
```

Comment:
What if the primary key of this table scan already has the best locality? Do we still recommend index here?

Reply:
Also, do we still send a notice when user is actually going to create an index?

Reply:
> What if the primary key of this table scan already has the best locality? Do we still recommend index here?

Yes. We check it inside `notice_recommended_index`.

Reply:
No. Because we only recommend an index when there is no indexes could be used, but once users create the index we recommend, the index must be able to be used.